grpc/creds/creds_test.go

```diff
@@ -0,0 +1,106 @@
+package creds
+
+import (
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"math/big"
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/letsencrypt/boulder/test"
+)
+
+func slowHandler(w http.ResponseWriter, r *http.Request) {
+	time.Sleep(time.Millisecond * 200)
+}
+
+func TestTransportCredentials(t *testing.T) {
+	serverA := httptest.NewTLSServer(nil)
+	defer serverA.Close()
+	addrA := serverA.Listener.Addr().String()
+	serverB := httptest.NewTLSServer(nil)
+	defer serverB.Close()
+	addrB := serverB.Listener.Addr().String()
+
+	priv, err := rsa.GenerateKey(rand.Reader, 1024)
+	test.AssertNotError(t, err, "rsa.GenerateKey failed")
+
+	temp := &x509.Certificate{
+		SerialNumber: big.NewInt(1),
+		Subject: pkix.Name{
+			CommonName: "A",
+		},
+		NotBefore:             time.Unix(1000, 0),
+		NotAfter:              time.Now().AddDate(1, 0, 0),
+		BasicConstraintsValid: true,
+		IsCA: true,
+	}
+	derA, err := x509.CreateCertificate(rand.Reader, temp, temp, priv.Public(), priv)
+	test.AssertNotError(t, err, "x509.CreateCertificate failed")
+	certA, err := x509.ParseCertificate(derA)
+	test.AssertNotError(t, err, "x509.ParserCertificate failed")
+	temp.Subject.CommonName = "B"
+	derB, err := x509.CreateCertificate(rand.Reader, temp, temp, priv.Public(), priv)
+	test.AssertNotError(t, err, "x509.CreateCertificate failed")
+	certB, err := x509.ParseCertificate(derB)
+	test.AssertNotError(t, err, "x509.ParserCertificate failed")
+	// XXX: don't do this
```

Comment:
This is a kind of hacky way of initializing the test TLS servers I meant to fix before pushing.